pre-check/v1.x/check.sh

```diff
@@ -0,0 +1,544 @@
+    fi
+}
+
+check_error_pods()
+{
+    log_info "Starting Pod Status check..."
+
+    pods=$(kubectl get pods -A -o yaml)
+    no_ok=$(echo "$pods" | yq '.items | any_c(.status.phase != "Running" and .status.phase != "Succeeded")')
+
+    if [ "$no_ok" = "false" ]; then
+        log_verbose "All pods are OK."
+        log_info "Pod-Status Test: Pass"
+        echo -e "\n==============================\n"
+        return
+    fi
+
+    log_info "There are non-ready pods:"
+    log_info "$(echo "$pods" | yq '.items[] | select(.status.phase != "Running" and .status.phase != "Succeeded") | .metadata.namespace + "/" + .metadata.name')"
+    record_fail "Pod-Status"
+}
+
+# Only works in Harvester nodes.
+check_free_space()
+{
+    log_info "Starting Node Free Space check..."
+    prom_ip=$(kubectl get services/rancher-monitoring-prometheus -n cattle-monitoring-system -o yaml | yq -e '.spec.clusterIP')
+    log_verbose "Trying to get results from ${prom_ip}:9090"
+    result=$(curl -sg "http://$prom_ip:9090/api/v1/query?query=node_filesystem_avail_bytes{mountpoint=\"/usr/local\"}<32212254720" | jq '.data.result')
+    if [ -z "${result}" ]; then
+        log_info "Script wasn't able to get valid response from the API.\nYou may need to log into each of the nodes and run 'df -h /usr/local' to ensure there's more than 30 GB of free space available."
+        log_verbose "Note: This check requires that the script is running from a control-plane node and that rancher-monitoring Addon is enabled."
+        record_fail "Node-Free-Space"
+        return
+    fi
+    length=$(echo "$result" | jq 'length')
+
+    if [ "$length" == "0" ]; then
+        log_verbose "All nodes have more than 30GB free space."
+        log_info "Node-Free-Space Test: Pass"
+        echo -e "\n==============================\n"
+        return
+    fi
+
+    log_info "Nodes doesn't have enough free space:"
+    log_info "$(echo "$result" | jq -r '.[].metric.instance')"
+    record_fail "Node-Free-Space"
+}
+
+#If a log file exists ask users if they want to clear it, or exit. 
+check_log_file()
+{
+    if [ -e "$log_file" ]; then
+        # Just clear the file if run with -y
+        if [ "$answer" = true ]; then
+            echo -e "[$(date +'%Y-%m-%d %H:%M:%S')] Cleared Previous Log File" > "$log_file"
+            return
+        fi
+        read -r -p "The file $log_file exists. Are you sure that you want to overwrite/clear the contents of that file? [Y/N]" response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])$ ]]; then
+            echo "Cleared Stale Log File" > $log_file
+            return
+        else
+            echo -e "Choose a different filename for the log flag (-l) or move/rename the file and run the script again. \nExiting."
+            exit 1
+        fi
+    fi
+}
+
+# Are we running the script on a control-plane node?
+check_host()
+{
+    log_info "Starting Host check... "
+    cp_nodes=$(kubectl get nodes |grep control-plane |awk '{ print $1 }')
```

Comment:
Let's say I have a 2-node Harvester cluster. One of the nodes is the control-plane node, and the other is the worker node. When I execute the check script on the worker node, the line `kubectl get nodes...` will have a problem because there's no kubeconfig file on the system. As I mentioned earlier, we have `-e` specified at the very beginning of the script. The execution will bail out immediately if it encounters an error.